app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -967,10 +983,9 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
                     if (!lastValue.toString().equals(getString(chooseDateOptionStringId))) {
                         dateField.setText(lastValue);
                     }
-                    DialogFragment datePickerFragment = new DatePickerFragment(
-                            LoyaltyCardEditActivity.this,
-                            dateField,
-                            // if the expiry date is being set, set date picker's minDate to the 'valid from' date
+                    DialogFragment datePickerFragment = DatePickerFragment.newInstance(
+                            loyaltyCardField,
+                            (Date) dateField.getTag(),
                             loyaltyCardField == LoyaltyCardField.expiry ? (Date) validFromField.getTag() : null,
```

Comment:
Looks like you accidentally removed one comment too much

```suggestion
                            // if the expiry date is being set, set date picker's minDate to the 'valid from' date
                            loyaltyCardField == LoyaltyCardField.expiry ? (Date) validFromField.getTag() : null,
```